packages/pretty-format/src/index.js

```diff
@@ -405,13 +456,13 @@ function prettyFormat(val, opts) {
     indent = createIndent(opts.indent);
     refs = [];
     const pluginsResult = printPlugin(val, indent, prevIndent, spacing, edgeSpacing, refs, opts.maxDepth, currentDepth, opts.plugins, opts.min, opts.callToJSON, opts.printFunctionName, opts.escapeRegex, colors);
-    if (pluginsResult) {
+    if (pluginsResult !== null) {
       return pluginsResult;
     }
   }
 
   const basicResult = printBasicValue(val, opts.printFunctionName, opts.escapeRegex);
-  if (basicResult) {
+  if (basicResult !== null) {
```

Comment:
Yes, I’m glad that you see and ask about this.

* Because `printBasicValue` doesn’t return undefined or empty string, there’s no change in behavior for the internal code path, as verified by tests.

* Because the 3 current built-in serializers don’t return those, no change there either.

* The reason to recommend an **intentional change** in behavior now before many people write serializers, is to allow a serializer to return **empty string**.

  ```js
  exports[`Allow user-defined serializer to return empty string 1`] = ``
  ```

  With current logic, `print` cannot distinguish no serializer`test` method matches from a serializer `test` method does match and its `print` method returns empty string.

  Although a rare edge case, I ran into the problem in first attempt at diff serializer, when it returned empty string for no differences between prev and next state.

* Yes, I agree that `!= null` is even better, in case a user-defined plugin returns undefined.